src/SimpleThings/EntityAudit/AuditReader.php

```diff
@@ -748,12 +748,24 @@ public function getEntityValues($className, $entity)
     {
         $metadata = $this->em->getClassMetadata($className);
         $fields = $metadata->getFieldNames();
+        $associationsFields = $metadata->getAssociationNames();
 
         $return = array();
         foreach ($fields AS $fieldName) {
             $return[$fieldName] = $metadata->getFieldValue($entity, $fieldName);
         }
 
+        foreach($associationsFields as $key => $association) {
+            $associationValues = $entity->{'get' . ucfirst($association)}();
```

Comment:
Didn't know about `$metadata->getFieldValue`. It's the same behavior as property access but for doctrine. Sounds the way to go.